plugins/ltac/extratactics.ml4

```diff
@@ -544,21 +544,6 @@ END
 
 
 
-(**********************************************************************)
-(*spiwack : Vernac commands for retroknowledge                        *)
-
-VERNAC COMMAND EXTEND RetroknowledgeRegister CLASSIFIED AS SIDEFF
- | [ "Register" constr(c) "as" "int31" pre_ident(n) ] ->
-           [ let env = Global.env () in
-             let evd = Evd.from_env env in
-             let tc,_ctx = Constrintern.interp_constr env evd c in
-             let tc = EConstr.to_constr evd tc in
-             let f = Retroknowledge.(KInt31 (int31_field_of_string n)) in
-             Global.register f tc ]
-END
-
-
-
```

Comment:
Why moving this into the hardcoded vernac AST?

Reply:
@ppedrot where would you put it? It clearly does not belong to `extratactics`.

Reply:
Dunno, but I think I prefer using extensions rather than hardcoding vernaculars. The former tends to lead to a more uniform syntax, as there is no way to cheat the parser or the printer, which usually happens spontaneously with hardcoded ASTs.

Reply:
I'm not sure what you mean exactly. But in any case the priority is to fix the location. This command has nothing to do with Ltac. If you want to change the "hard coded" AST for one based on extensions (although I'm not sure it would be a good change), it should be done separately.

Now, if there's a place where this command could belong under its extension form, just let us know and we'll move it there.

Reply:
That's not a hard requirement, essentially nitpicking. There is no point in delaying this welcome cleanup.